lib/updatehub/updatehub.c

```diff
@@ -33,6 +33,11 @@ LOG_MODULE_REGISTER(updatehub);
 #include <net/tls_credentials.h>
 #endif
 
+#if defined(CONFIG_WATCHDOG)
+/* implement this function in another module if watchdog is used */
+extern void watchdog_kick(void);
```

Comment:
Just pondering here aloud. Can we provide a default implementation of this kicker with __weak linkage? Then if needed, application would override that by providing an implementation for the kicker.

Reply:
This sounds like a better solution! In this case, do you suggest something like this?
    __weak void updatehub_watchdog_kick(void) {};

Reply:
yes, that is what I meant

Reply:
Thank you for the input. I rewrote it to use __weak linkage now.